shuup/admin/modules/media/static_src/media/browser/BrowserView.js

```diff
@@ -100,7 +100,18 @@ export function controller(config={}) {
     };
     ctrl.reloadFolderTree = function() {
         remote.get({"action": "folders"}).then(function(response) {
-            ctrl.rootFolder(response.rootFolder);
+            if (response.rootFolder.name === "Root" && !response.rootFolder.canSeeRoot) {
+                // Hide the root folder if the user cannot access it.
+                // This is only a visual thing to avoid showing the empty folder.
+                // The views handles the actual permission checking and clears the whole file
+                // QueyrSet of the root folder if the user is not allowed to access it.
+                const currentFolder = response.rootFolder.children[0];
+                ctrl.rootFolder(currentFolder || {});
```

Comment:
This will actually most likely result in any sibling folders of `currentFolder` to be completely inaccessable...

Reply:
IMO, these folders shouldn't even come from the API. Can this expose things we don't want to?